### PR TITLE
Added style parameter

### LINKED ISSUE #705: Add a way for different styled badges
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes
| RFC         | yes
| BC Break    | no

#### Summary

I'd like to be able to specify different styles for badges, namely `plastic`, `flat`, `flat-squared` and (maybe) `for-the-badge`.

| style | example 
|---|---
| `plastic` | image
| `flat` | image
| `flat-squared` | image
| `for-the-badge` | image

I think some are already possible, but it's not documented anywhere and I was unable to find any through trial and error. What do you think?

### PR DESCRIPTION
**Pull request by ricardoboss**
This PR will partially fix #705.

Why partially? Because [badges/poser](https://github.com/badges/poser) doesn't support the fourth requested style: `for-the-badge`. There is already an issue for this: [badges/poser#55](https://github.com/badges/poser/issues/55). I will try and implement this over there as well.

### CODE REVIEW

**Comment by JellyBellyDev** [comment]
@ricardoboss thanks so much for your PR! What do you think to add also the specific section on the landing page to explain the new feature?

**Comment by ricardoboss** [comment]
Sounds reasonable. I will add some documentation to this PR.

**Comment by ricardoboss** [comment]
@JellyBellyDev I need some help adding the `style` parameter to the routes. What I did does not seem to work... Can you take a look?

File: src/Controller/Badge/VersionController.php [hunk 1]
```diff
@@ -32,17 +33,21 @@ public function version(
...
     ): Response {
-        if (\in_array($request->query->get('format'), $poser->validStyles(), true)) {
-            $format = $request->query->get('format');
-        }
+//        if (!\in_array($request->query->get('style'), $poser->validStyles(), true)) {
+//            $style = Badge::DEFAULT_STYLE;
+//        }
+
+        var_dump($style);
+        die;
```
Comment (JellyBellyDev) [inline src/Controller/Badge/VersionController.php:44]: die

File: templates/home/index.html.twig [hunk 2]
```diff
@@ -104,6 +104,29 @@
...
+                                data-badge="{{ badge.name }}"
+                                readonly
+                                type="text"
+                                value="{{ badge.markdown }}" title="{{ badge.label }}">
+                        <button data-clipboard-target=".badge-input[data-badge='{{ badge.name }}']">Copy</button>
+                    </div>
+                {% endfor %}
+            </div>
+        </div>
+    </section>
```
Comment (JellyBellyDev) [inline templates/home/index.html.twig:128]: can you put a screenshot please! :)

File: src/Controller/Badge/VersionController.php (hunk 1 above)
Comment (ricardoboss) [reply src/Controller/Badge/VersionController.php:44]: This is for testing. I am not getting the actual style parameter I put in the URL. It does not work yet! Can you help?

File: templates/home/index.html.twig (hunk 2 above)
Comment (ricardoboss) [reply templates/home/index.html.twig:128]: Will do once I get the correct badges rendered :)

File: src/Controller/Badge/VersionController.php (hunk 1 above)
Comment (JellyBellyDev) [reply src/Controller/Badge/VersionController.php:44]: after making the change, restart docker-compose, sometimes the changes are not applied in the container :S

File: src/Badge/Model/UseCase/CreateComposerLockBadge.php [hunk 3]
```diff
@@ -55,13 +55,13 @@ public function __construct(
      * @throws UnexpectedValueException
      * @throws \GuzzleHttp\Exception\GuzzleException
      */
-    public function createComposerLockBadge(string $repository, string $format = 'svg'): CacheableBadge
+    public function createComposerLockBadge(string $repository, string $format = 'svg', string $style = 'flat'): CacheableBadge
```
Comment (AlessandroMinoccheri) [inline src/Badge/Model/UseCase/CreateComposerLockBadge.php:58]: We can use a constant for `flat` or other values?

Comment (ricardoboss) [reply src/Badge/Model/UseCase/CreateComposerLockBadge.php:58]: We could use `App\Badge\Model\Badge::DEFAULT_STYLE`. I didn't to keep the signatures consistent. If we refactor this, I would suggest also replacing the default value of `$format` with `App\Badge\Model\Badge::DEFAULT_FORMAT`

Comment (AlessandroMinoccheri) [reply src/Badge/Model/UseCase/CreateComposerLockBadge.php:58]: I agree @ricardoboss

File: src/Controller/Badge/VersionController.php (hunk 1 above)
Comment (JellyBellyDev) [reply src/Controller/Badge/VersionController.php:44]: as soon as I have a second I throw an eye too ;)

**Comment by JellyBellyDev** [comment]
Hi @ricardoboss,
I tagged a new feat on https://github.com/badges/poser.
Can you update this PR? Thanks

**Comment by ricardoboss** [comment]
@JellyBellyDev done

**Comment by ricardoboss** [comment]
No, because I haven't updated the PUGX/poser package yet.

File: templates/home/index.html.twig (hunk 2 above)
Comment (JellyBellyDev) [reply templates/home/index.html.twig:128]: WDYT? :)

Schermata 2022-02-18 alle 18 12 07

File: src/Controller/Badge/VersionController.php (hunk 1 above)
Comment (JellyBellyDev) [reply src/Controller/Badge/VersionController.php:44]: I fixed your problem! Can i commit on this branch?

File: templates/home/index.html.twig (hunk 2 above)
Comment (ricardoboss) [reply templates/home/index.html.twig:128]: First the flat variants, then plastic and for-the-badge. Speaking of, can you update PUGX/poser to also include the fourth style, please? :D

File: src/Controller/Badge/VersionController.php (hunk 1 above)
Comment (ricardoboss) [reply src/Controller/Badge/VersionController.php:44]: Sure

File: templates/home/index.html.twig (hunk 2 above)
Comment (JellyBellyDev) [reply templates/home/index.html.twig:128]: I tried to do composer update package but I've a error! :S

Comment (JellyBellyDev) [reply templates/home/index.html.twig:128]:
```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires badges/poser ^2.3 -> satisfiable by badges/poser[v2.3.0].
    - badges/poser v2.3.0 requires kartsims/easysvg ^2.0 -> found kartsims/easysvg[dev-master, 1.0] but it does not match the constraint.
```

File: src/Controller/Badge/VersionController.php (hunk 1 above)
Comment (JellyBellyDev) [reply src/Controller/Badge/VersionController.php:44]: I add my commit here: https://github.com/PUGX/badge-poser/pull/733

File: templates/home/index.html.twig (hunk 2 above)
Comment (ricardoboss) [reply templates/home/index.html.twig:128]: We need to add my fork as a repository as well, it seems...

```
    "repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/ricardoboss/easysvg"
        }
    ],
```

**Comment by ricardoboss** [comment]
@JellyBellyDev do you want me to close this PR in favor of #733?

**Comment by JellyBellyDev** [comment]
Yea! Thanks! in doing so we can work together! ;)

File: src/Controller/Badge/VersionController.php (hunk 1 above)
Comment (ricardoboss) [reply src/Controller/Badge/VersionController.php:44]: Why exactly did you open a second PR? You should be able to commit to my fork.

**Pull request closed without merging**